File: CSharp/Mayope.Api/Types/Forest.cs
using System.Collections.Generic;

namespace Mayope.Api.Types;
{
    /// <summary>
    /// Details of a forest.
    /// </summary>
    public class Forest : Node
    {
        /// <summary>
        /// The children of this forest; empty if none, null if this is a stub
        /// record.
        /// </summary>
        public IList<Forest> Forests { get; set; }

        /// <summary>
        /// The trees within this forest; empty if none, null if this is a stub
        /// record.
        /// </summary>
        public IList<Tree> Trees { get; set; }
    }
}

File: Java/org/mayope/api/types/Forest.java
package org.mayope.api.types;

import java.util;

/**
 * Details of a forest.
 */
public class Forest extends Node {
    /**
     * The children of this forest; empty if none, null if this is a stub
     * record.
     */
    private List<Forest> forests;

    /**
     * The trees within this forest; empty if none, null if this is a stub
     * record.
     */
    private List<Tree> trees;

    /**
     * Get the children of this forest; empty if none, null if this is a stub
     * record.
     *
     * @return The value
     */
    public List<Forest> getForests() {
        return forests;
    }

    /**
     * Set the children of this forest; empty if none, null if this is a stub
     * record.
     *
     * @param forests The new value
     */
    public void setForests(List<Forest> forests) {
        this.forests = forests;
    }

    /**
     * Get the trees within this forest; empty if none, null if this is a stub
     * record.
     *
     * @return The value
     */
    public List<Tree> getTrees() {
        return trees;
    }

    /**
     * Set the trees within this forest; empty if none, null if this is a stub
     * record.
     *
     * @param trees The new value
     */
    public void setTrees(List<Tree> trees) {
        this.trees = trees;
    }
}
